eng/common/pipelines/templates/steps/cosmos-emulator.yml

```diff
@@ -9,8 +9,84 @@ steps:
       Write-Host "Target Dir: $targetDir"
       msiexec /a ${{ parameters.EmulatorMsiUrl }} TARGETDIR=$targetDir /qn | wait-process
     displayName: Download and Extract Public Cosmos DB Emulator
+  - powershell: |
+      Write-Host "Deleting Cosmos DB Emulator data"
+      if (Test-Path $Env:LOCALAPPDATA\CosmosDbEmulator) { Remove-Item -Recurse -Force $Env:LOCALAPPDATA\CosmosDbEmulator }
+    displayName: Delete Cosmos DB Emulator data
+  - powershell: |
+      Write-Host "Getting Cosmos DB Emulator Version"
+      $ProductName = "Azure Cosmos DB Emulator"
+      $Emulator = (Join-Path $env:temp (Join-Path $ProductName "Microsoft.Azure.Cosmos.Emulator.exe"))
+      $fileVersion = Get-ChildItem $Emulator
+      Write-Host $Emulator $fileVersion.VersionInfo
+    displayName: Get Cosmos DB Emulator Version
   - powershell: |
       Write-Host "Launching Cosmos DB Emulator"
-      Import-Module "$env:temp\Azure Cosmos DB Emulator\PSModules\Microsoft.Azure.CosmosDB.Emulator"
```

Comment:
@kushagraThapar does the cosmos module not work any longer?

Reply:
@weshaggard - they don't .. I had that conversation with you sometime last month, where we were seeing older versions of cosmos emulator in the azure VMs. So we came up with this script to always install the latest version, since there are some new features in latest version that our code relies on now.

Reply:
Or may be it was with Mitch, can't remember, but yes, this was an issue that we found out where cosmos emulator version on the Azure VMs was old, and we couldn't control which version of emulator Azure VMs bring in. They update it every once in a while, but not instantly. So we needed our own script.

Reply:
This was the PR where we first introduced this script - https://github.com/Azure/azure-sdk-for-java/pull/18056/files#r567032679
And then it got overwritten.

Reply:
OK I'm not super familiar with the cosmos emulator modules but if we are essentially reimplementing them inline here I highly suggest we pull the changes out into a script file so it is easier to run and debug locally. Having a lot of inline code like this makes diagnosing very difficult.

As for the change that got overwritten. We have a particular process of any changes that go under eng/common, that Dozie linked to https://github.com/Azure/azure-sdk-tools/blob/master/doc/common/common_engsys.md. Any changes in there are subject to being overwritten because it is a one-way sync from the tools repo to ensure that all the common tools/scripts are consistent across the different languages.

For these changes I assume this emulator change will work for all the languages and not just java correct? If not we need to also need make the changes to make those work consistently. As part of getting this PR merged into the tools repo we will need to run the cosmos tests in each of the 4 primary language repos in the sync PRs for this change to ensure that these work for those.

Reply:
Thanks for the explanation @weshaggard, yes I believe it should work for all languages. @FabianMeiswinkel @milismsft  any reason why this change might not work for other languages in case my understanding is not correct ?

Reply:
Different languages might use different arguments which control how the Cosmos DB Emulator is started, such as different physical partition count for instance. These arguments have direct impact as to what features are enabled and can be tested, the startup performance of the emulator etc; one setting that fits all it is not practical in this case.

@weshaggard I really have hard time understanding this process of having two separate repos controlling same set of files with no good practices when it comes to sync'ing/merging changes from to another. May be we should completely remove the whole eng/common/pipeline out of this repo, stick a "readme" file instead as where any pipeline related changes should go into what repo. And we would also be better if we have some way to configure those based on the language specifics.

BTW even as of now, there's no solution to practically reproduce a CI issue locally because the images used by the CI are not available to an ordinary user like myself. I will recommend tackling and fixing that first, before we think of the debugging and scripting of these PowerShell scripts.

Reply:
@milismsft the files under "eng/common" are not supposed to be edited directly in the language repos and we do have a README in that directory calling that out https://github.com/Azure/azure-sdk-for-java/blob/master/eng/common/README.md. We always mirror those files from our tools repo to all the different language repos because they are intended to be shared between all the language repos.

If there are language specific settings or changes they shouldn't be under the eng/common directory, but instead should be somewhere else in the language repository. For this case if the scripts are mostly common we should share as much as we can across the different language teams and provide options to tweak and language specific settings in the language repo's when they consume this template.

Even if we cannot easily run the emulator exactly like we do in CI it is still a good idea to be able to have a script so that we can debug the script itself without running a full CI build to test and find simile code issues in the script.